tests/unit/workflows/test_workflows_tasks_submission.py

```diff
@@ -22,22 +22,380 @@
 
 from __future__ import absolute_import, division, print_function
 
-from inspirehep.modules.workflows.tasks.submission import add_note_entry
+from inspirehep.modules.workflows.tasks.submission import (
+    submit_rt_ticket,
+    create_ticket,
+    reply_ticket,
+    close_ticket,
+    send_robotupload,
+    add_note_entry,
+    filter_keywords,
+    prepare_keywords,
+    user_pdf_get,
+    prepare_files,
+    remove_references,
+)
 
+from six import StringIO
+from mock import MagicMock
+import mock, pytest
 
-class StubObj(object):
-    def __init__(self, data, extra_data):
+
+class MockObj(object):
+    class MockLog(object):
+        def __init__(self):
+            self._log = StringIO()
+
+        def info(self, *args, **kwargs):
+            self._log.write(args[0])
+
+        def error(self, *args, **kwargs):
+            self.info(args[0])
+
+        def debug(self, message, format=None):
+            self.info(message)
+
+    log = MockLog()
+
+    def __init__(self, id_, data, extra_data, collection_file={}):
+        self.id = id_
+        self.id_user = id_
         self.data = data
         self.extra_data = extra_data
+        self.files = collection_file
+
+
+def mock_context_function(user, obj):
+    return {}
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.get_instance')
+@mock.patch('inspirehep.modules.workflows.tasks.submission.in_production_mode')
+def test_submit_rt_ticket(mock_production_mode, mock_rt_instance):
+    mock_file = MagicMock()
+    mock_file.create_ticket = MagicMock(return_value=1)
+    mock_rt_instance.return_value = mock_file
```

Comment:
I would expect that a variable called `mock_file` is the mock of a file, but you clearly mean something different. What do you mean here?